src/dmv.py

```diff
@@ -34,66 +62,192 @@ class CA_DMV():
             "plateChar6": "",
         }
     PLATE_CHAR_PREFIX = "plateChar"
-    MIN_LENGTH = 2
-    MAX_LENGTH = 7
+
+    UPPERCASE_INPUT = True
+    MUST_MATCH_PATTERNS = [(re.compile(pattern), reason) for pattern, reason in [
+        ("^[A-Z1-9/ ]{2,7}$", "Must contain 2 to 7 letters, numbers (1-9), and spaces (' ' or '/')."),
+        ("[A-Z1-9].*[A-Z1-9]", "Must contain at least 2 non-space characters."),
```

Comment:
one at the beginning and one at the end ? can't start or end in spaces? what about /?

Reply:
Notice that the regex isn't surrounded by `^` and `$`, and we're doing `search`, not `match`. So this pattern isn't enforcing start/end. just existence of two non-space characters somewhere in the string.

`/` is a half-space character.